xtext/org.icyphy.linguafranca/src/lib/core/rti.h

```diff
@@ -153,34 +153,71 @@ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 #define TIMED_MESSAGE 5
 
-/** Byte identifying a next event time (NET) message sent from a federate.
- *  The next eight bytes will be the timestamp. This message from a
- *  federate tells the RTI the time of the earliest event on that federate's
- *  event queue. In other words, absent any further inputs from other federates,
- *  this will be the logical time of the next set of reactions on that federate.
+/** 
+ * Byte identifying a next event tag (NET) message sent from a federate.
+ * The next eight bytes will be the timestep.
+ * The next four bytes will be the microstep.
+ * This message from a federate tells the RTI the tag of the earliest event 
+ * on that federate's event queue. In other words, absent any further inputs 
+ * from other federates, this will be the logical time of the next set of
+ * reactions on that federate.
  */
 #define NEXT_EVENT_TIME 6
 
-/** Byte identifying a time advance grant (TAG) sent to a federate.
- *  The next eight bytes will be the timestamp.
+/** 
+ * Byte identifying a time advance grant (TAG) sent to a federate.
+ * The next eight bytes will be the timestamp.
+ * The next four bytes will be the microstep.
  */
 #define TIME_ADVANCE_GRANT 7
 
-/** Byte identifying a logical time complete (LTC) message sent by a federate
- *  to the RTI. The next eight bytes will be the timestamp.
+/** 
+ * Byte identifying a logical tag complete (LTC) message sent by a federate
+ * to the RTI.
+ * The next eight bytes will be the timestep of the completed tag.
+ * The next four bytes will be the microsteps of the completed tag.
  */
 #define LOGICAL_TIME_COMPLETE 8
 
-/** Byte identifying a stop message. When any federate calls stop(), it will
- *  send this message to the RTI, which will then broadcast it to all other
- *  federates. The next 8 bytes will be the timestamp.
- *  NOTE: It is not clear whether sending a stopping timestamp is useful.
- *  If any federate can send a STOP message that specifies the stop time on
- *  all other federates, then every federate depends on every other federate
- *  and time cannot be advanced. Hence, the current implementations may result
- *  in nondeterministic stop times.
+/////////// Messages used in request_stop() ///////////////
```

Comment:
How does the implemented solution relate to https://github.com/icyphy/lingua-franca/issues/221 and https://github.com/icyphy/lingua-franca/wiki/Termination?

Reply:
Timeout, stop request, and external signal interrupt should all follow the specification in [Termination](https://github.com/icyphy/lingua-franca/wiki/Termination). I have not yet tackled the problem of starvation. The true distributed versions of the aforementioned features are still not implemented as well. For now, I have assumed that the RTI is in charge of coordinating termination.